servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/AmbSingles.java

```diff
@@ -78,6 +78,7 @@ protected void handleSubscribe(final Subscriber<? super T> subscriber) {
 
     static final class AmbSubscriber<T> extends DelayedCancellable implements Subscriber<T> {
         private final State<T> state;
+        private volatile boolean propagateCancel = true;
```

Comment:
We have an opportunity to avoid the second `volatile` state inside `AmbSubscriber` if we consolidate it under current `Cancellable` from `DelayedCancellable`. We can do this either by adding a new `protected` method `DelayedCancellable.ignoreCancel()` that will set `current` to `IGNORE_CANCEL` or (if we don't want more API on `DelayedCancellable`) we can remove `extends DelayedCancellable` from `AmbSubscriber` and repeat its logic here with a new feature to ignore cancel post termination.